In [1]:
# Import relevant Python libraries.
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from scipy import stats
import statsmodels.api as sm


## Load data

In [2]:
epa_data = pd.read_csv('c4_epa_air_quality.csv', index_col=0)
epa_data.head()

date_local    state_name   county_name      city_name  \
0  2018-01-01       Arizona      Maricopa        Buckeye   
1  2018-01-01          Ohio       Belmont      Shadyside   
2  2018-01-01       Wyoming         Teton  Not in a city   
3  2018-01-01  Pennsylvania  Philadelphia   Philadelphia   
4  2018-01-01          Iowa          Polk     Des Moines   

                                     local_site_name   parameter_name  \
0                                            BUCKEYE  Carbon monoxide   
1                                          Shadyside  Carbon monoxide   
2  Yellowstone National Park - Old Faithful Snow ...  Carbon monoxide   
3                             North East Waste (NEW)  Carbon monoxide   
4                                          CARPENTER  Carbon monoxide   

    units_of_measure  arithmetic_mean  aqi  
0  Parts per million         0.473684    7  
1  Parts per million         0.263158    5  
2  Parts per million         0.111111    2  
3  Parts per million         0.300000    3  
4  Parts per million         0.215789    3

In [8]:
epa_data = epa_data.dropna()

In [3]:
# Get descriptive stats.
epa_data.describe()

arithmetic_mean         aqi
count       260.000000  260.000000
mean          0.403169    6.757692
std           0.317902    7.061707
min           0.000000    0.000000
25%           0.200000    2.000000
50%           0.276315    5.000000
75%           0.516009    9.000000
max           1.921053   50.000000

- The count value for the `aqi` column is 260. This means there are 260 aqi measurements represented in this dataset.

- The 25th percentile for the `aqi` column is 2. This means that 25% of the aqi values in the data are below 2.

- The 75th percentile for the `aqi` column is 9. This means that 75% the aqi values in the data are below 9.

In [4]:
# Get descriptive stats about the states in the data.
epa_data['state_name'].describe()

count            260
unique            52
top       California
freq              66
Name: state_name, dtype: object

- There are 260 states values, and 52 of them are unique. California is the most commonly occurring state in the data, with a frequency of 66. 

In [5]:
# Compute the mean value from the aqi column.
np.mean(epa_data['aqi'])

6.757692307692308

In [6]:
# Compute the median value from the aqi column.
np.median(epa_data['aqi'])

5.0

In [7]:
print(epa_data['aqi'].min())
print(epa_data['aqi'].max())
print(epa_data['aqi'].std())


0
50
7.061706678820724
